tests/pjit_test.py

```diff
@@ -372,7 +372,7 @@ def _test_rule(*args, **kwargs):
       xla._translations[pjit_p] = rule
 
   @jtu.with_mesh([('x', 2)])
-  def testLowerWithAbstractArgs(self):
+  def testLowerWithDuckTyping(self):
```

Comment:
I think that's already tested here, i.e. this test case is using `lower` with `ShapeDtypeStruct`. I just renamed this test.

Reply:
Ah, I missed the `lower` call on the last line of the test. Still, similar to the `jit`-related test in this PR, should we also `compile` and check the output of what's compiled?

Reply:
I dno, seems like it could be redundant. The duck typing part should be fully exercised just by `lower`.

So if you're okay with it, I'd rather just land!